enterprise/ha/src/main/java/org/neo4j/kernel/ha/HaSettings.java

```diff
@@ -47,9 +47,14 @@
     @Description( "How long a slave will wait for response from master before giving up." )
     public static final Setting<Long> read_timeout = setting( "ha.read_timeout", DURATION, "20s" );
 
-    @Description( "Timeout for waiting for instance to become master or slave." )
+    @Description( "Timeout for request threads waiting for instance to become master or slave." )
     public static final Setting<Long> state_switch_timeout = setting( "ha.state_switch_timeout", DURATION, "120s" );
 
+    @Description( "Timeout for waiting for internal conditions during state switch, like for transactions "
+            + "to complete, before switching to master or slave." )
+    public static final Setting<Long> internal_state_switch_timeout =
+    setting( "ha.internal_state_switch_timeout", DURATION, "10s" );
```

Comment:
Odd formatting. I'd line-break after `setting(` and indent the parameters. Is 10 seconds not quite long, with a default heartbeat timeout of 11 seconds?